### PR TITLE
Support Query Insights

### PR DESCRIPTION
**Pull request by itkq**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/8328

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
sql: added `insights_config` block to `google_sql_database_instance` resource
```

### CODE REVIEW

**Comment by itkq** [comment]
@googlebot I signed it!

**Review by ve6yeq — APPROVED** [review]
LGTM

**Comment by c2thorn** [comment]
Sorry for the delay, I should be able to get to this over the next day or so

File: mmv1/third_party/terraform/website/docs/d/sql_database_instance.html.markdown [hunk 1]
```diff
@@ -179,3 +179,13 @@ The `replica_configuration` block contains:
 * `server_ca_cert.0.expiration_time` - Expiration time of the CA Cert.
 
 * `server_ca_cert.0.sha1_fingerprint` - SHA Fingerprint of the CA Cert.
+
+The `settings.insights_config` subblock for instances declares [Query Insights](https://cloud.google.com/sql/docs/postgres/insights-overview) configuration. It contains:
```
Comment (c2thorn) [inline mmv1/third_party/terraform/website/docs/d/sql_database_instance.html.markdown:183]: Can we move this sub block just above the `replica_configuration` block (line 129)? To be closer with other `settings` blocks

File: mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb [hunk 2]
```diff
@@ -1322,6 +1346,26 @@ resource "google_sql_database_instance" "instance" {
 }
 `
 
+var testGoogleSqlDatabaseInstance_insights = `
+resource "google_sql_database_instance" "instance" {
+  name   = "tf-lw-%d"
```
Comment (c2thorn) [inline mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb:1351]:
```suggestion
  name   = "tf-test-%d"
```
Allows any hanging database instances to be swept (some other test configs in this file are outdated)

**Review by c2thorn — CHANGES_REQUESTED** [review]
Only a couple of small asks here, overall great! Tests also passing.

File: mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb (hunk 2 above)
Comment (itkq) [reply mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb]: Fixed https://github.com/GoogleCloudPlatform/magic-modules/pull/4464/commits/42df0e23f30b878825c04b7849efd259e47aaccd

File: mmv1/third_party/terraform/website/docs/d/sql_database_instance.html.markdown (hunk 1 above)
Comment (itkq) [reply mmv1/third_party/terraform/website/docs/d/sql_database_instance.html.markdown]: Agreed https://github.com/GoogleCloudPlatform/magic-modules/pull/4464/commits/8464a2023f375aa7e28ad1400a38b4fcc3b88ed2

**Review by c2thorn — APPROVED** [review]

**Comment by itkq** [comment]
Thanks!

**Pull request merged**